### PR TITLE
[TIMOB-23114] iOS: Fix semi-bold and bold fonts-weights

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23114

### CODE REVIEW

**Review by ewieberappc — APPROVED** [review]
FR Passed, using:
MacOS 10.12.6 (16G24b)
Studio 4.9.0.201705302345
Ti SDK 7.0.0.v20170809111003
Appc NPM 4.2.9
Appc CLI 6.2.3
Alloy 1.9.13
Xcode 8.3.3 (8E3004b)

Font weight now includes semibold. Bold and italics are working as expected. Semibold fonts can also be used by their postscript name.

File: iphone/Classes/WebFont.m [hunk 1]
```diff
@@ -151,9 +151,9 @@ -(UIFont *) font
                 }
             }
             if (font == nil) {
-                //NO valid family specified. Just check for characteristics. Semi bold is ignored here.
+                //NO valid family specified. Just check for characteristics.
                 if (self.isBoldWeight) {
-                    UIFont* theFont = [UIFont boldSystemFontOfSize:self.size];
+                    UIFont* theFont = [UIFont systemFontOfSize:self.size weight:UIFontWeightBold];
```
Comment (vijaysingh-axway) [inline iphone/Classes/WebFont.m]: This API is available in iOS 8.2 and above only. Need to put check for same.
+ (UIFont *)systemFontOfSize:(CGFloat)fontSize weight:(CGFloat)weight NS_AVAILABLE_IOS(8_2);

File: iphone/Classes/WebFont.m [hunk 2]
```diff
@@ -180,14 +180,16 @@ -(UIFont *) font
                     }
                 } else if (self.isItalicStyle) {
                     font = [[UIFont italicSystemFontOfSize:self.size] retain];
+                } else if (self.isSemiboldWeight) {
+                    font = [[UIFont systemFontOfSize:self.size weight:UIFontWeightSemibold] retain];
```
Comment (vijaysingh-axway) [inline iphone/Classes/WebFont.m]: Same as above.

File: iphone/Classes/WebFont.m (hunk 1 above)
Comment (hansemannn) [inline iphone/Classes/WebFont.m]: I didn't on purpose, since we target 8.2 and later since SDK 6.0.0.

Comment (hansemannn) [inline iphone/Classes/WebFont.m]: Actually, we do 8.0 I guess. Maybe we can find out the raw-value and pass that one instead - for backwards-compatibility?

Comment (eric34) [inline iphone/Classes/WebFont.m]: @hansemannn  We could get it in before 7.0.0 but I am ok with pushing it out of 6.2.0 with a decent comment.

Comment (hansemannn) [inline iphone/Classes/WebFont.m]: Let's do it in 7.0.0, I am trying not to risk anything here.

**Comment by hansemannn** [comment]
Fixed as part of #9308

**Pull request closed without merging**